adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DefaultDeviceResolver.java

```diff
@@ -103,6 +116,13 @@ private static SecretKey getKey(final JsonObject candidateSecret) {
         }
     }
 
+    private Span newSpan(final String operation) {
+        return tracer.buildSpan(operation)
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
```

Comment:
if these spans are suppose to be used for tracking the retrieval of the PSK from the device registry, then FMPOV the kind is `SPAN_KIND_CLIENT`.

Reply:
FMPOV, two spans are used for that. The outer one describes the incoming dtls handshake message (dtls server), the inner, the "getCredentials" call (amqp client). So, aligned to the POST stuff, I chose "server" for the outer and "client" for the inner.

Reply:
Can you point me to the place where the outer Span for tracing the DTLS handshake is created? I am afraid that I couldn't find it.

Reply:
FMPOV that are the callbacks 

```
@Override
public SecretKey getKey(final PskPublicInformation identity)
 
@Override
public AdditionalInfo getInfo(final Principal clientIdentity)
```